lib/sycamore/sycamore/connectors/common.py

```diff
@@ -70,6 +70,23 @@ def flatten_data(
     return items
 
 
+def unflatten_data(data: dict[str, Any], separator: str = ".") -> dict[str, Any]:
+    result: dict[Any, Any] = {}
+
+    for key, value in data.items():
+        parts = key.split(separator)
+        current = result
+
+        for i, part in enumerate(parts):
+            if i == len(parts) - 1:
+                current[part] = value
+            else:
+                if part not in current:
+                    current[part] = {}
```

Comment:
if the next part is a number this should maybe be a list? idk. I'm okay not dealing with unflattening lists rn

Reply:
Makes sense, have updated it